puppet/modules/eayunstack/manifests/upgrade/ceilometer/ceilometer.pp

```diff
@@ -87,6 +116,29 @@
       enable => true,
     }
 
+    file { 'pipeline.yaml':
+      path => '/etc/ceilometer/pipeline.yaml',
+      ensure => file,
+      source => 'puppet:///modules/eayunstack/pipeline.yaml',
+      group => 'ceilometer',
+      require => Package['openstack-ceilometer-common'],
+      notify => Service['openstack-ceilometer-compute'],
+    }
+
+    augeas { 'ceilometer-pipeline':
+      context => '/files/etc/ceilometer/ceilometer.conf',
+      lens => 'Puppet.lns',
+      incl => '/etc/ceilometer/ceilometer.conf',
+      changes => [
+        "set DEFAULT/pipeline_cfg_file /etc/ceilometer/pipeline.yaml",
+      ],
+      require => Package['openstack-ceilometer-common'],
+      notify => Service['openstack-ceilometer-compute'],
+    }
+
+    File['pipeline.yaml'] ~>
```

Comment:
我已将File['pipeline.yaml'] 合并到require 里面了．

2016-04-01 12:54 GMT+08:00 huntxu <notifications@github.com>:

> In puppet/modules/eayunstack/manifests/upgrade/ceilometer/ceilometer.pp
> <https://github.com/eayunstack/eayunstack-upgrade/pull/26#discussion_r58162397>
> :
>
> > +      require => Package['openstack-ceilometer-common'],
> > +      notify => Service['openstack-ceilometer-compute'],
> > +    }
> > +
> > +    augeas { 'ceilometer-pipeline':
> > +      context => '/files/etc/ceilometer/ceilometer.conf',
> > +      lens => 'Puppet.lns',
> > +      incl => '/etc/ceilometer/ceilometer.conf',
> > +      changes => [
> > +        "set DEFAULT/pipeline_cfg_file /etc/ceilometer/pipeline.yaml",
> > +      ],
> > +      require => Package['openstack-ceilometer-common'],
> > +      notify => Service['openstack-ceilometer-compute'],
> > +    }
> > +
> > +    File['pipeline.yaml'] ~>
>
> 这里也一样建议将这个文件放到上面定义的require中就好，不需要用~>
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/eayunstack/eayunstack-upgrade/pull/26/files/10b8d0f96412cf4b09c9fd499d9f85e865b71d07#r58162397>
>